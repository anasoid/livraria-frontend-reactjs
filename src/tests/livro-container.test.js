import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import LivroContainer from '../components/livro/livro-container';

describe('test LivroConteiner component', () => {

  const wrapper = shallow(<LivroContainer/>);

  it('should renders without crashing ', () => {
    shallow(<LivroContainer/>);
    const tree = renderer.create(<LivroContainer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be "Novo Livro" h1 title ', () => {
    expect(wrapper.find('h1')).toHaveText('Novo Livro');
  });

  it('should be input titulo do livro', () => {
    const titulo = wrapper.find('#livroTitulo');
    expect(titulo).toMatchSelector('input');
    expect(titulo).toHaveProp('name', 'titulo');
  });

  it('should be input isbn do livro', () => {
    const isbn = wrapper.find('#livroIsbn');
    expect(isbn).toMatchSelector('input');
    expect(isbn).toHaveProp('name', 'isbn');
  });


});