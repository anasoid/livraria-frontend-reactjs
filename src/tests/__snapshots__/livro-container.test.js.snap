// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LivroConteiner component should renders without crashing  1`] = `
<div>
  <h1>
    Novo Livro
  </h1>
  <label
    htmlFor="livroTitulo"
  >
    Titulo: 
  </label>
  <input
    id="livroTitulo"
    name="titulo"
  />
  <br />
  <label
    htmlFor="livroIsbn"
  >
    ISBN: 
  </label>
  <input
    id="livroIsbn"
    name="isbn"
  />
</div>
`;
